sdk/python/feast/data_source.py

```diff
@@ -598,7 +563,7 @@ def __init__(
         self._file_options = FileOptions(file_format=file_format, file_url=file_url)
 
         super().__init__(
-            event_timestamp_column or self._infer_event_timestamp_column(r"^timestamp"),
+            event_timestamp_column or "",
```

Comment:
can this just be `None` instead of an empty string?

Reply:
yep can delete the "or ""' part. Just need to change default value for event_timestamp_column of base class (DataSource) to None instead of "". Was hesitant to change base class param since didn't know if the use of "" was intentional.

Reply:
oh actually, it seems like the data source base class doesn't have event_timestamp_column as optional so its type is str instead of Option[str]. The subclasses constructors use Option[str] so when passing to the supers' constructor need to handle the type checking and `or ""` turns the type from a Option[str] to str. I'll add comments to make it clear this is for type checking.

Reply:
Could we make the `event_timestamp_column` optional in the base data source class? I think that would be reasonable if possible.

Reply:
yep can do. Was thinking that not all subclasses of DataSource base class is guaranteed to have inference so maybe from an interface point of view, it's good to communicate that DataSource actually needs a event_timestamp_column.

Reply:
resolved on slack. Achal - "I think that the subclasses can add additional assertions as per their requirements, but the base class should be more broad". Will make the change